### ISSUE TITLE
nomad init creates job that doesn't match anything in the docs (very bad UX)

### ISSUE THREAD

**Issue by apollo13** [hashicorp/nomad#8746]
### Nomad version
Nomad v0.12.3 (2db8abd9620dd41cb7bfe399551ba0f7824b3f61)

### Issue
`./nomad init -short` generates a job file that does not match stanzas from the documentation

### Reproduction steps
```
$ ./nomad init -short 
Example job file written to example.nomad
$ cat example.nomad 
job "example" {
  datacenters = ["dc1"]

  group "cache" {
    task "redis" {
      driver = "docker"

      config {
        image = "redis:3.2"

        port_map {
          db = 6379
        }
      }

      resources {
        cpu    = 500
        memory = 256

        network {
          mbits = 10
          port "db" {}
        }
      }
    }
  }
}
```

Now `job -> group -> task -> resources` (https://www.nomadproject.io/docs/job-specification/resources) does not mention `network` at all and `job -> group -> network` (https://www.nomadproject.io/docs/job-specification/network) doesn't mention that it is allowed in `resources` either. I understand that task level networking is deprecated with the move to groups and consul connect, but either the example should use documented syntax.

**Comment by tgross** [comment]
Hi @apollo13! It looks like we hadn't managed to update that yet. We've marked this as a bug to fix but if you're up to it, we'd welcome a PR to the [asset files](https://github.com/hashicorp/nomad/tree/master/command/assets) where that job is defined!

**Comment by apollo13** [comment]
@tgross I tried that somewhat in #8747 -- I am honestly confused about *how* it should look like nowadays :)

**Comment by nickethier** [comment]
Hey @apollo13 I have some fixes for this slated to go in #8743 

Would you mind taking a look and letting me know if theres still any confusion I could further clear up?

**Comment by apollo13** [comment]
@nickethier Fails with:
```
    2020-09-01T09:23:27.828+0200 [ERROR] client.alloc_runner.task_runner: running driver failed: alloc_id=756f5cf5-8aa8-58c2-169b-0cb22066332a task=redis error="Failed to create container configuration for image "redis:3.2" ("sha256:87856cc39862cec77541d68382e4867d7ccb29a85a17221446c857ddaebca916"): Trying to map ports but no network interface is available"
```

**Comment by nickethier** [comment]
fixed in #8911
